environment/src/test/java/com/sequenceiq/environment/credential/validation/CredentialValidatorTest.java

```diff
@@ -46,24 +54,77 @@
 
     @BeforeEach
     void setUp() {
-        underTest = new CredentialValidator(Set.of("AWS", "AZURE"), credentialDefinitionService, Collections.emptyList(), entitlementService);
+        underTest = new CredentialValidator(Set.of(AWS, AZURE), credentialDefinitionService, Collections.emptyList(), entitlementService);
     }
 
-    @Test
-    void testValidateCredentialCloudPlatformWhenBadPlatform() {
-        assertThrows(BadRequestException.class, () -> underTest.validateCredentialCloudPlatform("FOO", USER_CRN));
+    // @formatter:off
+    // CHECKSTYLE:OFF
+    static Object[][] validateCredentialCloudPlatformWhenAzureDisabledDataProvider() {
+        return new Object[][] {
+                // testCaseName cloudPlatform   validExpected
+                { AWS,          AWS,            true },
+                { AZURE,        AZURE,          false },
+                { FOO,          FOO,            false },
+        };
+    }
+    // CHECKSTYLE:ON
+    // @formatter:on
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("validateCredentialCloudPlatformWhenAzureDisabledDataProvider")
+    void testValidateCredentialCloudPlatformWhenAzureDisabled(String testCaseName, String cloudPlatform, boolean validExpected) {
+        if (AZURE.equalsIgnoreCase(cloudPlatform)) {
+            when(entitlementService.azureEnabled(USER_CRN, ACCOUNT_ID)).thenReturn(false);
```

Comment:
the Azure enabled/disabled can be a parameter of this ParameterizedTest ;)

Reply:
Indeed. Done.